doc/tutorial/index.rst

```diff
@@ -437,13 +437,74 @@ And make the sentence you added in ``index.rst`` look like this:
    :ref:`install <installation>` the project.
 
 Notice a trick here: the ``install`` part specifies how the link will look like
-(we want it to be a specific word, so the sentence makes sense), whereas the
-``<installation>`` part refers to the actual label we want to add a
+(in this case, using a specific word so the sentence makes sense), whereas the
+``<installation>`` part refers to the actual label you want to add a
 cross-reference to. If you do not include an explicit title, hence using
 ``:ref:`installation```, the section title will be used (in this case,
 ``Installation``). Both the ``:doc:`` and the ``:ref:`` roles will be rendered
 as hyperlinks in the HTML documentation.
 
+Referencing external projects with intersphinx
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You are likely depending on some open source libraries for your own
+project, and there is a chance those dependencies have their documentation
+written in Sphinx and available online. One of the most powerful features of
+Sphinx is the ability to generate cross-references to other
+projects available online, using the
+:doc:`intersphinx </usage/extensions/intersphinx>` extension.
+
+To enable it, first add ``sphinx.ext.intersphinx`` to the list of extensions:
+
+.. code-block:: python
+   :caption: docs/source/conf.py
+
+   extensions = [
+       'sphinx.ext.duration',
+       'sphinx.ext.intersphinx',
+   ]
+
+The way to know whether you can reference an external project is to check
+if it has a publicly available ``objects.inv``, which can be parsed calling
+``python -m sphinx.ext.intersphinx``. For example, this is the output for
+Sphinx itself:
+
+.. code-block:: console
+   :emphasize-lines: 4
+
+   (.venv) $ python -msphinx.ext.intersphinx https://www.sphinx-doc.org/en/master/objects.inv  | grep 'std:doc' --after-context 5
+   std:doc
+           changes                                  Changelog                               : changes.html
+           contents                                 Sphinx documentation contents           : contents.html
+           development/builders                     Configuring builders                    : development/builders.html
+           development/index                        Extending Sphinx                        : development/index.html
+           development/overview                     Developing extensions overview          : development/overview.html
+
+To link your documentation with the Sphinx one, add a
+:confval:`intersphinx_mapping` dictionary to your ``conf.py`` as follows:
+
+.. code-block:: python
+   :caption: docs/source/conf.py
+
+   intersphinx_mapping = {
+       'sphinx': ('https://www.sphinx-doc.org/en/master/', None),
+   }
+
+You can now introduce a cross-reference to the ``contents`` section of the
+Sphinx documentation as follows, optionally prefixing it by ``sphinx:``, the
```

Comment:
This prefixing is a bit of an icky issue. It doesn't work in general for all roles (and fundamentally can't), so in the end this should illustrate the feature with the role introduced by #9062.

Reply:
I was not aware of that pull request. Do we want to introduce it even if it's not merged yet? Also, out of curiosity, can you expand a bit (or give an example) that doesn't work?

Reply:
The tutorial should definitely be in sync with the implementation, so it's more that we should be aware of updating this part when relevant. Though, I guess this part will be merged after 4.1 (going out this weekend I believe), so there is a chance of the PR being ready in the same release as this.
As concrete examples, the roles in the C and C++ domains do not support this prefix. There are more details in https://github.com/sphinx-doc/sphinx/pull/8418#issuecomment-758838589, but the fundamental issue is that when you call a role, e.g., ``:py:class:`` you ask the Python domain extension to do whatever it likes. This is most likely to insert a ``pending_xref`` node with meta-information, which that same extension is being asked to resolve at a later stage. When you suddenly add the prefix, you explicitly don't want that domain to handle it, but hope it understands and fails to resolve the xref, such that intersphinx can get a chance to resolve according to the prefix. Hence, fundamentally the syntax should be with a role that directly calls intersphinx directly, without other extensions involved (until a later stage, see also #8929).

Reply:
As I commented above, one of the goal of intersphinx is transparency. So I prefer not to use prefix for cross references. As you explained, it is "optional" feature. So it would be better to explain the extension by non-prefixed example at first, I think.

Reply:
Fair enough, it looked more clar to me that way, but I'm +1 on explaning the fully consistent and transparent way instead. Will rephrase this part.